dso/model.py

```diff
@@ -239,73 +209,77 @@ class FilesystemPath:
 
 @dataclasses.dataclass(frozen=True)
 class StructureInfo:
-    id: BDBAPackageId
-    scan_id: BDBAScanId
+    package_name: str
+    package_version: str | None # bdba might be unable to determine a version
     licenses: list[License]
     filesystem_paths: list[FilesystemPath]
 
     @property
     def key(self) -> str:
-        return f'{self.id.key}|{self.scan_id.key}'
+        return f'{self.package_name}|{self.package_version}'
 
 
 @dataclasses.dataclass(frozen=True)
 class Finding:
-    id: BDBAPackageId
-    scan_id: BDBAScanId
+    '''
+    Base class for artefact metadata which is interpreted as a finding
+    '''
```

Comment:
please also state that childs must implement `key` property (to uniquely identify a finding within its domain?)

Reply:
However, the `key` property should not only be implemented by derived classes from `Finding` but by all possible dataclasses for the `data` property or not?

Reply:
Added a doc string to `ArtefactMetadata` model class 👍🏼